### DISCUSSION TITLE
Add page like functionality to home buttons

### DISCUSSION THREAD

**Discussion by vm2700** [nplan/HomeButtons#68] [Q&A]
I have the original 6 button model.  I would like to devote button 6 to be a "Next Page" button.  Every time you press button 6, buttons 1 - 5 will change their labels and their functionality.  That way I could have 3-4 pages of buttons and each button could trigger 3-4 different commands.

Has anyone already implemented something like this?

Another question: Is it possible to change the font size of the labels to something smaller?

A third question - can I have just the status (temp, humidity, battery) display until any button is pressed?

**Comment by nplan** [comment]
1. The pages functionality is not implemented at the moment.
2. To use smaller font you can prepend the label with _.
3. Not possible at the moment. Are you thinking the first press would show the buttons and the second press would trigger action?

**Reply by vm2700** [reply]
Thanks for the quick response.  I managed to implement the pages functionality.  I did it by having button 6 increment a global variable (I call the variable pgnum) and change all the labels of the buttons.  When you press any other button, it implements the appropriate action based on the current value of pgnum.  I can cycle through as many pages as I want.

Thanks for answering my second question.

For my third question, yes, I was thinking that touching any button will bring up the current button labels and the second touch would actually take an action.

**Reply by vm2700** [reply]
For the third question, in general one could implement an information display that is updated periodically with useful information (hourly weather, bus schedule etc).  When you press any button the information screen goes away and the buttons are displayed for several seconds.

**Reply by vm2700** [reply]
The disp_msg command will display a message but the "\n" for new lines does not appear to work.  Also, is it possible to get an even smaller font?  Something like 12-15 chars/line would allow a fair amount of information to be displayed.

**Reply by nplan** [reply]
I might implement the information display functionality in the future.

The '\n' does not work if you input it in the Home Assistant text input box. But it should work in automations and service call YAML.

**Reply by vm2700** [reply]
OK, thanks.
